### PR TITLE
removes named function support (NFN)

### LINKED ISSUE #177: remove NFN code in ccn-lite
This is a follow up from #161 

Tasks:
* Remove NFN-related code from ccn-lite code base
* Update documentation

### LINKED ISSUE #234: remove nfnexpr parameter from function ccnl_URItoPrefix
Remove the parameter ``char *nfnexpr`` from function ``ccnl_URItoPrefix``, i.e. change signature from
``` C
struct ccnl_prefix_s *ccnl_URItoPrefix(char* uri, int suite, 
   char *nfnexpr, unsigned int *chunknum);
```
to 
``` C
struct ccnl_prefix_s *ccnl_URItoPrefix(char* uri, int suite, unsigned int *chunknum);
```

This issue is related to #177

### PR DESCRIPTION
**Pull request by mfrey**
This PR removes the named function support (NFN) from ccn-lite. I strongly recommend to test it extensively. This PR also breaks the ccn-lite shell commands in RIOT (see also ``sys/shell/commands/scn_ccnl.c`` since the ``nfnexpr`` parameter was removed from ``ccnl_URItoPrefix``). A subsequent PR is going to be made in RIOT.

It's not clear to me how to handle the docker image and the ``*.deb`` and ``*.rpm`` support. Also, a few figures in the documentation reference NFN which have not been changed (yet).

This PR (obviously) requires squashing. I've split everything in small commits so reviewing (hopefully) and rolling back (when necessary) gets easier.

### Contribution description

 Basically, code blocks guarded by `` USE_NFN``, ``USE_NFN_REQUESTS``, ``USE_TIMEOUT_KEEPALIVE``, and ``USE_NACK`` have been removed. Also documentation has been removed or customized where necessary.

### Issues/PRs references

This PR fixes #177. It also fixes #234 and depends on #161.

### CODE REVIEW

**Comment by blacksheeep** [comment]
I think we should freeze the code and issue a last release of "ccn-lite with nfn and nfn scala" in case somebody still wants to use the code.
@mfrey i will need some time to run tests and to test different situations. 
@cgundogan i think we should fix that in riot before merging?

**Comment by mfrey** [comment]
> @mfrey i will need some time to run tests and to test different situations.

Absolutely! If I can be of any help, please let me know!

> @cgundogan i think we should fix that in riot before merging?

I would disagree. RIOTs package management system points to a specific commit (so it does not rely on the current HEAD of a specific branch). Hence, we merge the PR at some point in the future and RIOT's version of ccn-lite points to an old commit (so nothing breaks). A subsequent PR in RIOT points to the new version (updating the ``PKG_VERSION`` variable) and fixes the (broken) calls to ``ccnl_URItoPrefix``

For a better understanding see RIOTs current version (i.e. the [Makefile](https://github.com/RIOT-OS/RIOT/blob/master/pkg/ccn-lite/Makefile)) of the ccn-lite package:
```
PKG_NAME=ccn-lite
PKG_URL=https://github.com/cn-uofbasel/ccn-lite/
PKG_VERSION=dbf3e1a8935cfc4b6bcfa1a5baa477553ffdafec
PKG_LICENSE=ISC

.PHONY: all
...
```

**Comment by blacksheeep** [comment]
I am ok with freezing the current state, I tested it recently and it worked. 
Nevertheless, I did not performed enough tests for merging this PR, this may take some time to verify.

**Comment by mfrey** [comment]
> Nevertheless, I did not performed enough tests for merging this PR, this may take some time to verify.

I'll try to address the merge conflicts introduced with the recently merged PRs.

**Comment by mfrey** [comment]
> I'll try to address the merge conflicts introduced with the recently merged PRs.

Done.
